ui/Screen/Onboarding/EnterName.svelte

```diff
@@ -1,74 +1,36 @@
 <script>
   import { createEventDispatcher } from "svelte";
-  import validatejs from "validate.js";
 
-  import {
-    ValidationStatus,
-    getValidationState,
-  } from "../../src/validation.ts";
+  import { ValidationStatus } from "../../src/validation.ts";
+  import * as onboarding from "../../src/onboarding.ts";
 
   import { Button, Emoji, Input } from "../../DesignSystem/Primitive";
 
-  export let handle = null;
-
   const dispatch = createEventDispatcher();
 
-  const HANDLE_MATCH = "^[a-z0-9][a-z0-9_-]+$";
+  export let handle = null;
 
-  let validations,
-    beginValidation = false;
+  let beginValidation = false;
+  const validationStore = onboarding.createHandleValidationStore();
 
-  validatejs.options = {
-    fullMessages: false,
-  };
+  $: beginValidation && validationStore.validate(handle);
+  $: allowNext = (handle && validationPasses()) || !validationStarted();
```

Comment:
Why is `validationStarted` needed? If no handle is provided, it shouldn't `allowNext` and if one is provided, but `validationPasses` returns `false` it should not `allowNext`, ie if the `validationStore.status` is anything other than `ValidationStatus.Success`

Reply:
If we don't have that check, then the button stays disabled forever.

Reply:
hm ok, I also couldn't figure it out. Seems repetitive somehow